networkx/readwrite/json_graph/adjacency.py

```diff
@@ -149,9 +149,9 @@ def adjacency_graph(data, directed=False, multigraph=True, attrs=_attrs):
             target = target_data.pop(id_)
             if not multigraph:
                 graph.add_edge(source, target)
-                graph[source][target].update(tdata)
+                graph[source][target].update(target_data)
```

Comment:
IIUC this will add the info *without* the id attribute - what's the reason for that? One good way to show this would be to add a test for this behavior that fails on `main` and passes after this change.

Reply:
The id attribute is the target of the edge, currently serialising e.g. 

```G = DiGraph()
G.add_edge("A", "B")
H = adjacency_graph(adjacency_data(G))
H.edges[("A", "B")]  # = {"id": "B"}
G.edges[("A", "B")]  # = {}
```

Which means that graphs_equal(G, H) == False.
The id attribute isn't required or wanted here.

node_link does a similar thing with ```edgedata = {str(k): v for k, v in d.items() if k != source and k != target}``` (although it could stand to be more obviously analogous by using pop)